### PR TITLE
Fix "ReturnValueIgnored" errors flagged by error-prone

### PR DESCRIPTION
**Pull request by kluever**

### CODE REVIEW

**Comment by bhuvanaindukuri** [comment]
Which version of java were you able to replicate this issue? Also isn't it better to replace the line 
`e.getMessage();` with either `assertTrue(true);` or just remove the statement altogether as done in other methods like "testAssertNullNotEqualsString"

**Review by bhuvanaindukuri — CHANGES_REQUESTED** [review]
Which version of java were you able to replicate this issue? Also isn't it better to replace the line
e.getMessage(); with either assertTrue(true); or just remove the statement altogether as done in other methods like "testAssertNullNotEqualsString"

File: src/test/java/junit/tests/framework/AssertTest.java [hunk 1]
```diff
@@ -82,15 +82,15 @@ public void testAssertStringNotEqualsNull() {
             assertEquals("foo", null);
             fail();
         } catch (ComparisonFailure e) {
-            e.getMessage(); // why no assertion?
+            String unused = e.getMessage(); // why no assertion?
```
Comment (kcooney) [inline src/test/java/junit/tests/framework/AssertTest.java:85]: I suggest we use `assertNotNull()` here and below.

**Review by kcooney — CHANGES_REQUESTED** [review]

File: src/test/java/junit/tests/framework/AssertTest.java (hunk 1 above)
Comment (kluever) [reply src/test/java/junit/tests/framework/AssertTest.java]: Sure, done.

**Comment by kluever** [comment]
> Which version of java were you able to replicate this issue?

At Google, we've enabled CheckReturnValue by default, so all non-void APIs that are not annotated with `@CanIgnoreReturnValue` must have their values used. This has caught thousands of bugs so far, and we hope to be open sourcing this soon!

**Review by kcooney — APPROVED** [review]

**Review by bhuvanaindukuri — APPROVED** [review]
Approving this change but it would be nice to make this change in other tests as well where there is nothing in the catch block.

**Comment by kcooney** [comment]
@bhuvanaindukuri if you send me a pull request doing that I would approve it. Alternately we could have the code assert that the message contained a string.

Note these tests are very old. They verify the JUnit 3.x assertion code

**Comment by kcooney** [comment]
Asking @marcphilipp or @sbrannen to provide a second approval, then I can merge.

**Review by sbrannen — APPROVED** [review]
LGTM

**Pull request merged**